pystxmcontrol/utils/writeNX.py

```diff
@@ -517,292 +364,70 @@ def addFrame(self,frame,framenum, mode="dark"):
-        ds.attrs["version"] = nxFileVersion
-        entry_nxgrp.create_dataset(name='title', data=Title, maxshape=None)
-        entry_nxgrp.create_dataset(name='start_time', data=[Stime], maxshape=None)
-        entry_nxgrp.create_dataset(name='end_time', data=[Etime], maxshape=None)
-        entry_nxgrp.create_dataset(name='proposal', data=[Title], maxshape=None)
-        entry_nxgrp.create_dataset(name='experimenters', data=[Experimenters], maxshape=None)
-        entry_nxgrp.create_dataset(name='sample_description', data=[Sample], maxshape=None)
-        motor_grp = _group(entry_nxgrp, 'motors', NXmonitor)
-
-        #
-        #	The control group
-        #
-        control_nxgrp = _group(entry_nxgrp, 'control', NXmonitor)
-        _list_attr(control_nxgrp, 'axes', np.string_(['energy', 'sample_y', 'sample_x']))  # Attributes of entry1/control
-        _list_attr(control_nxgrp, 'energy_indices', [0])
-        _list_attr(control_nxgrp, 'sample_x_indices', [2])
-        _list_attr(control_nxgrp, 'sample_y_indices', [1])
-        _list_attr(control_nxgrp, 'signal', np.string_(['data']))
-
-        #		The datasets under control
-        _dset_arb_attr(control_nxgrp, 'data', Cu3D, 'mA')
-        _dset_arb_attr(control_nxgrp, 'energy', e, 'eV', '/entry1/counter0/energy')
-        _dset_arb_attr(control_nxgrp, 'sample_x', x, 'um', '/entry1/counter0/sample_x')
-        _dset_arb_attr(control_nxgrp, 'sample_y', y, 'um', '/entry1/counter0/sample_y')
-        _dset_arb_attr(control_nxgrp, 'sample_z', z, 'um', '/entry1/counter0/sample_z')
-        #
-        #	entry1/counter0 group, with attributes
-        #
-        c0_nxgrp = _group(entry_nxgrp, 'counter0', NXdata)
-        _list_attr(c0_nxgrp, 'axes', np.string_(['energy', 'sample_y', 'sample_x']))
-        _list_attr(c0_nxgrp, 'energy_indices', [0])
-        _list_attr(c0_nxgrp, 'sample_x_indices', [2])
-        _list_attr(c0_nxgrp, 'sample_y_indices', [1])
-        _list_attr(c0_nxgrp, 'signal', np.string_(['data']))
-
-        #		The datasets under entry1/counter0
-        _dset_arb_attr(c0_nxgrp, 'data', Counts, '')
-
-        if self.scan_dict['spiral'] and 'Image' == self.scan_dict["type"]:
-            _dset_arb_attr(c0_nxgrp, 'count_time', DAQ_Dwells, 's')
-
-        _dset_arb_attr(c0_nxgrp, 'energy', e, 'eV', '/entry1/counter0/energy')
-        _dset_arb_attr(c0_nxgrp, 'sample_x', x, 'um', '/entry1/counter0/sample_x')
-        _dset_arb_attr(c0_nxgrp, 'sample_y', y, 'um', '/entry1/counter0/sample_y')
-        _dset_arb_attr(c0_nxgrp, 'sample_z', z, 'um', '/entry1/counter0/sample_z')
-        c0_nxgrp.create_dataset(name='stxm_scan_type', data=ScanType, maxshape=None)
 
+    def createEntry(self,i):
+        ne,nz_m,ny_m,nx_m = len(self.energies),len(self.zMeasured[i]),len(self.yMeasured[i]),len(self.xMeasured[i])
+        nz_r,ny_r,nx_r = len(self.zPos[i]), len(self.yPos[i]), len(self.xPos[i])
+
+        nxentry = self.NXfile.create_group("entry%i" %i)
```

Comment:
another opportunity for new string format:

`        nxentry = self.NXfile.create_group(f'entry{i}')
`